packages/clay-css/gulpfile.js

```diff
@@ -231,11 +227,22 @@ gulp.task('version', function() {
 		}
 	});
 
-	fs.readdirSync(svgsDir).forEach(function(file) {
-		var filePath = path.join(svgsDir, file);
+	var libIconsSvg = path.join('.', 'lib', 'images', 'icons', 'icons.svg');
+
+	var regex = /<\?xml version=\"1.0\" encoding=\"UTF-8\"\?([\s\S]+-->|>)/g;
```

Comment:
Pro-Tip™: you can use [the `s` flag](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/dotAll) to make `.` match everything, including newlines, which means you can get rid of the `[\s\S]` construct:


```js
var regex = /<\?xml version=\"1.0\" encoding=\"UTF-8\"\?(.+-->|>)/gs;
```

Also, you have some excess escaping in there:

```js
var regex = /<\?xml version="1.0" encoding="UTF-8"\?(.+-->|>)/gs;
```

Reply:
Thanks for the tips! Regex is a mystery to me. I think I understand `\s+` below, but I don't understand the difference between `.*` and `.+`. `.*` matches 0 or more of the preceding token and `.+` matches 1 or more of the preceding token. Where would `.*` match something I wouldn't want in this example vs `.+`?

Reply:
Original: `/\*\sClay\s(.*)\n/g`

Would match: `* Clay 4.0.0\n` but also `* Clay \n`

Proposed: `/\*\s+Clay\s(.+)\n/g`

Would only match: `* Clay 4.0.0\n`

In other words, the second one is more robust because it is less likely to run into a false positive if it sees a line like the second one here:

```
 * see also the documentation for
 * Clay 
```

Of course, we're talking about an unlikely edge case, but the art of this is being pragmatic: make stuff as robust as possible *at a reasonable cost*, but no more.

There's [that saying](https://en.wikipedia.org/wiki/Robustness_principle) about "be liberal in what you accept and conservative in what you provide", which, in the world of regexps, translates to something like this:

- Be "loose" so that you don't falsely reject inputs that could be valid. An example of that would be changing the `\s` to `\s*` or `\s+` in `/\*\s\Clay/`; ie. don't reject something on the basis of the amount of whitespace in an insignificant position.
- But be "strict" so that you don't accept obviously incorrect input. An example of that would be making something that matches version numbers instead of just `(.+)`. That could be something as simlpe as `/\d\.d\.\d/` or as complicated as `/(\d{1,2}\.\d{1,2}(?:\.\d{1,2})?(?:-.+)?)/`, or more...

Basically, it's all very subjective, and I think over time you develop a gut instinct for when it's time to take a step in the direction of strictness or of looseness, because you spider sense starts tingling about false positives or false negatives. There's also the trade-off on the axis of readability too (like those version number examples above); here again it's a largely subjective judgement about what is "optimal" — amusingly "optimal" doesn't mean "perfect", but rather "good enough". 😂